aten/src/ATen/native/cpu/DistanceOpsKernel.cpp

```diff
@@ -0,0 +1,205 @@
+#include "ATen/native/cpu/DistanceOpsKernel.h"
+
+#include <numeric>
+#include <iterator>
+#include <algorithm>
+
+#include "ATen/Dispatch.h"
+#include "ATen/Parallel.h"
+#include "ATen/cpu/vml.h"
+
+namespace at { namespace native { namespace {
+
+template<typename scalar_t>
+struct PDist {
+  using Vec = vec256::Vec256<scalar_t>;
+
+  static inline Vec sign(Vec val) {
+    return vec256::min(vec256::max(Vec(0), val.ceil()), Vec(1)) +
+      vec256::min(vec256::max(Vec(-1), val.floor()), Vec(0));
+  }
```

Comment:
This seems weirdly long for a sign-computing function. Is https://stackoverflow.com/questions/41315420/how-to-implement-sign-function-with-sse3 perhaps of interest? (We don't actually have to optimize it for this PR, but maybe a comment saying that it is not optimal might be reasonable).